pkg/landscaper/blueprints/blueprint.go

```diff
@@ -30,9 +30,46 @@ func New(blueprint *lsv1alpha1.Blueprint, content vfs.FileSystem) *Blueprint {
 		Info: blueprint,
 		Fs:   content,
 	}
+
+	// migration: set type field in import/export definitions if not set
+	// can be removed when the type field in import/export definitions is mandatory
+	migrateType(&b.Info.Imports)
```

Comment:
mhm not sure if we need that migration here.
I thought that we add the during the defaulting of blueprint (https://github.com/gardener/landscaper/blob/master/apis/core/v1alpha1/defaults.go#L17).

If we do it that we we would always have the type set which means we could always use the new code in the validation and in the handling. And would not require duplicated code or maintain old code path.

Reply:
Good point, I didn't find the correct place to add the defaulting, but that one looks good, will move it there.